packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -144,6 +187,64 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
         return false;
     }
 
+    private prepareVaultedInstrumentPaymentPayload(
+        methodId: string,
+        paypalOrderId: string,
+        paymentData: VaultedInstrument & HostedInstrument,
+    ): Payment {
+        const { instrumentId, shouldSetAsDefaultInstrument } = paymentData;
+
+        return {
+            methodId,
+            paymentData: {
+                formattedPayload: {
+                    vault_payment_instrument: null,
+                    set_as_default_stored_instrument: shouldSetAsDefaultInstrument || false,
+                    device_info: null,
+                    method_id: methodId,
```

Comment:
Is its necessary to provide method_id in formatted payload? Because as I see you are passing it as a part of the submit data.

Please check this mapping in bigpay-client-js repo. Thanks

Reply:
Yes, it seems there is no need to provide it. Thanks